rosidl_generator_cpp/rosidl_generator_cpp/__init__.py

```diff
@@ -198,13 +199,18 @@ def primitive_value_to_cpp(type_, value):
     if type_.typename in [
         'short', 'unsigned short',
         'char', 'wchar',
-        'double', 'long double',
         'octet',
         'int8', 'uint8',
         'int16', 'uint16',
     ]:
         return str(value)
 
+    if type_.typename in ['double', 'long double']:
+        if isnan(value):
```

Comment:
For some reason, I needed to surroudn `value` in a `float()` cast otherwise it threw.